app/models/lookup_value.rb

```diff
@@ -98,6 +98,18 @@ def ensure_matcher(match_type)
     matcher
   end
 
+  def ensure_matcher_exists
+    return false if match.blank?
+    match.split(LookupKey::KEY_DELM).each do |m|
+      key_element = m.split(LookupKey::EQ_DELM).first
+      unless lookup_key.path.include?(key_element)
+        errors.add(:match, _("%{key} does not exist in order field") % { :key => key_element })
+        return false
```

Comment:
No need to return here or at the end, adding the error's enough and lets the user see _all_ errors.